code/modules/reagents/Chemistry-Goon-420BlazeIt.dm

```diff
@@ -72,22 +100,40 @@ datum/reagent/crank/on_mob_life(var/mob/living/M as mob)
 	var/high_message = pick("You feel calm.", "You feel collected.", "You feel like you need to relax.")
 	if(prob(5))
 		M << "<span class='notice'>[high_message]</span>"
+	..()
+	return
+
+/datum/reagent/krokodil/overdose_process(var/mob/living/M as mob)
 	if(prob(10))
 		M.adjustBrainLoss(rand(1,5)*REM)
 		M.adjustToxLoss(rand(1,5)*REM)
-	if(cycle_count == 10)
-		M.adjustBrainLoss(rand(1,10)*REM)
-		M.adjustToxLoss(rand(1,10)*REM)
-	if(cycle_count == 20)
+	..()
+	return
+
+
+/datum/reagent/krokodil/addiction_act_stage1(var/mob/living/M as mob)
+	M.adjustBrainLoss(rand(1,5)*REM)
+	M.adjustToxLoss(rand(1,5)*REM)
+	..()
+	return
+/datum/reagent/krokodil/addiction_act_stage2(var/mob/living/M as mob)
+	if(prob(25))
 		M << "<span class='danger'>Your skin feels loose...</span>"
-	if(cycle_count == 50)
-		M << "<span class='userdanger'>Your skin falls off!</span>"
+	..()
+	return
+/datum/reagent/krokodil/addiction_act_stage3(var/mob/living/M as mob)
+	if(prob(25))
+		M << "<span class='danger'>Your skin starts to peel away...</span>"
+	M.adjustBruteLoss(3*REM)
+	..()
+	return
+/datum/reagent/krokodil/addiction_act_stage4(var/mob/living/M as mob)
+	if(!skin_fallen_off)
```

Comment:
Why make this a reagent var? sounds like it would make more sense as a mob var, or even better, a species type check (ie something like `if(istype(M.dna.species, /datum/species/human))`)